tests/test_basic.py

```diff
@@ -1,14 +1,96 @@
+import calendar
+import datetime
+from freezegun import freeze_time
 import golem_messages
-import golem_messages.message
+from golem_messages import exceptions
+from golem_messages import message
 import unittest
+import unittest.mock as mock
+
+one_second = datetime.timedelta(seconds=1)
+
+def dt_to_ts(dt):
+    return calendar.timegm(dt.utctimetuple())
 
 class BasicTestCase(unittest.TestCase):
     def setUp(self):
         self.ecc = golem_messages.ECCx(None)
         self.ecc2 = golem_messages.ECCx(None)
 
     def test_total_basic(self):
-        msg = golem_messages.message.MessagePing()
+        msg = message.MessagePing()
         payload = golem_messages.dump(msg, self.ecc.raw_privkey, self.ecc.raw_pubkey)
         msg2 = golem_messages.load(payload, self.ecc.raw_privkey, self.ecc.raw_pubkey)
         self.assertEqual(msg, msg2)
+
+testnow = datetime.datetime.utcnow().replace(microsecond=0)
+
+@freeze_time(testnow)
+class TimestampTestCase(unittest.TestCase):
+    """Time limits verification
+
+    Based on Concent_analiza_integracji_PL Limity czasu w komunikacji chapter
+    """
+
+    def setUp(self):
+        self.ecc = golem_messages.ECCx(None)
+        # mmtt - Maximum Message Transport Time, maksymalny dopuszczalny
+        #        czas na przesłanie małego komunikatu (jeśli ping przekracza
+        #        ten czas to komunikacja jest niedrożna).
+        self.mmtt = datetime.timedelta(minutes=0, seconds=30)
+        # mtd - Maximum Time Difference, maksymalne dopuszczalne odchylenie
+        #       czasu od czasu rzeczywistego.
+        self.mtd = datetime.timedelta(minutes=2, seconds=30)
+        # mat - Maximum Action Time, maksymalny czas na wykonanie prostej
+        #       operacji na maszynie.
+        self.mat = datetime.timedelta(minutes=2, seconds=15)
+
+    def test_timestamp_within_range(self):
```

Comment:
This contains 3 independent test cases, right? If so, they should be in separate methods.